app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt

```diff
@@ -0,0 +1,96 @@
+package com.example.pokedex.pokemonDetails
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.viewpager2.widget.ViewPager2
+import com.example.pokedex.R
+import com.example.pokedex.databinding.FragmentPokemonDetailsBinding
+import com.example.pokedex.pokemonDetails.pagerAdapter.PokemonDetailsPagerAdapter
+import com.example.pokedex.pokemonDetails.pagerFragments.MovesFragment
+import com.example.pokedex.pokemonDetails.pagerFragments.StatsFragment
+import com.example.pokedex.pokemonDetails.viewmodel.DetailsViewModel
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+import com.squareup.picasso.Picasso
+
+class PokemonDetailsFragment : Fragment() {
+    private val args: PokemonDetailsFragmentArgs by navArgs()
+    private var _binding: FragmentPokemonDetailsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var tabLayout: TabLayout
+    private lateinit var viewPager: ViewPager2
+    private lateinit var imageView: ImageView
+    private lateinit var nameTv: TextView
+    private lateinit var numberTv: TextView
+    private lateinit var detailsViewModel: DetailsViewModel
+    private var pokemonName: String = ""
+    private var pokemonNumber: Int = 0
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        detailsViewModel = ViewModelProvider(this)[DetailsViewModel::class.java]
+        pokemonName = args.pokemonName
+        pokemonNumber = args.pokemonNumber
+
+        _binding = FragmentPokemonDetailsBinding.inflate(inflater, container, false)
+        nameTv = binding.nameTV
+        nameTv.text = pokemonName
```

Comment:
you can have 
`binding.apply { 
nameTV = pokemon name
numberTV = formattedNumber

}`

All other binding related code can go in there too